esp-config/src/generate/mod.rs

```diff
@@ -45,8 +46,460 @@ impl fmt::Display for Error {
+            let result = (self.0).checked_neg();
+            if let Some(result) = result {
+                Ok(I128(result))
+            } else {
+                Err(EvalexprError::NegationError {
+                    argument: evalexpr::Value::<NumericTypes>::from_int(*self),
+                })
+            }
+        }
+
+        fn checked_mul(&self, rhs: &Self) -> EvalexprResult<Self, NumericTypes> {
+            let result = (self.0).checked_mul(rhs.0);
+            if let Some(result) = result {
+                Ok(I128(result))
+            } else {
+                Err(EvalexprError::MultiplicationError {
+                    multiplicand: evalexpr::Value::<NumericTypes>::from_int(*self),
+                    multiplier: evalexpr::Value::<NumericTypes>::from_int(*rhs),
+                })
+            }
+        }
+
+        fn checked_div(&self, rhs: &Self) -> EvalexprResult<Self, NumericTypes> {
+            let result = (self.0).checked_div(rhs.0);
+            if let Some(result) = result {
+                Ok(I128(result))
+            } else {
+                Err(EvalexprError::DivisionError {
+                    dividend: evalexpr::Value::<NumericTypes>::from_int(*self),
+                    divisor: evalexpr::Value::<NumericTypes>::from_int(*rhs),
+                })
+            }
+        }
+
+        fn checked_rem(&self, rhs: &Self) -> EvalexprResult<Self, NumericTypes> {
+            let result = (self.0).checked_rem(rhs.0);
+            if let Some(result) = result {
+                Ok(I128(result))
+            } else {
+                Err(EvalexprError::ModulationError {
+                    dividend: evalexpr::Value::<NumericTypes>::from_int(*self),
+                    divisor: evalexpr::Value::<NumericTypes>::from_int(*rhs),
+                })
+            }
+        }
+
+        fn abs(&self) -> EvalexprResult<Self, NumericTypes> {
+            Ok(I128(self.0.abs()))
+        }
+
+        fn bitand(&self, rhs: &Self) -> Self {
+            I128(std::ops::BitAnd::bitand(self.0, rhs.0))
+        }
+
+        fn bitor(&self, rhs: &Self) -> Self {
+            I128(std::ops::BitOr::bitor(self.0, rhs.0))
+        }
+
+        fn bitxor(&self, rhs: &Self) -> Self {
+            I128(std::ops::BitXor::bitxor(self.0, rhs.0))
+        }
+
+        fn bitnot(&self) -> Self {
+            I128(std::ops::Not::not(self.0))
+        }
+
+        fn bit_shift_left(&self, rhs: &Self) -> Self {
+            I128(std::ops::Shl::shl(self.0, rhs.0))
+        }
+
+        fn bit_shift_right(&self, rhs: &Self) -> Self {
+            I128(std::ops::Shr::shr(self.0, rhs.0))
+        }
+    }
+
+    impl evalexpr::EvalexprNumericTypes for I128NumericTypes {
+        type Int = I128;
+        type Float = f64;
+
+        fn int_as_float(int: &Self::Int) -> Self::Float {
+            int.0 as Self::Float
+        }
+
+        fn float_as_int(float: &Self::Float) -> Self::Int {
+            I128(*float as i128)
```

Comment:
This is probably not a good idea, we should use `truncate` on the float or something like that instead.